src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/NetworkSupervisor.kt

```diff
@@ -58,11 +66,15 @@ internal open class NetworkSupervisor(
     internal open fun didSetValidatorConnected(validatorConnected: Boolean) {
     }
 
+    internal open fun didSetValidatorUrl(validatorUrl: String?) {
```

Comment:
We can make `NetworkSupervisor` `abstract` instead of `open`. Then we can also make this method `abstract`.

`open` declares a concrete type that we can instantiate and want to extend. `abstract` declares an interface type. In this case, we want `abstract` because this is a function we want implementations to implement themselves.